src/main/java/org/spongepowered/api/text/dialogue/ChatDetails.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.dialogue;
+
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.channel.MessageChannel;
+import org.spongepowered.api.text.chat.ChatType;
+
+import java.util.Optional;
+
+/**
+ * A container class for use in {@link DialogueChatHandler}s.
+ */
+public interface ChatDetails {
+
+    /**
+     * Gets the message that was sent in chat.
+     *
+     * @return The message
+     */
+    Text getMessage();
+
+    /**
+     * Gets the {@link ChatType} that this message was sent under.
+     *
+     * @return The type of chat
+     */
+    ChatType getType();
+
+    /**
+     * Gets the channel that the chat was sent to.
+     *
+     * @return The channel
+     */
+    MessageChannel getChannel();
+
+    /**
+     * Gets whatever sent the message.
+     *
+     * @return The sender, if there was one
+     */
+    Optional<Object> getSender();
```

Comment:
@pie-flavor, To give you an example on this, 

`public` interface ChatDetails<T> extends TextRepresentable {

and...

`Optional<T> getSender();

If you wanted to ensure that T extends another class, such as CommandSource, you'd use:
`<T extends CommandSource>
only in the class declaration.